aten/src/ATen/core/function_schema_inl.h

```diff
@@ -42,20 +42,39 @@ inline std::ostream& operator<<(std::ostream& out, const FunctionSchema& schema)
   return out;
 }
 
-inline void FunctionSchema::checkArg(const IValue& value, const Argument& argument, optional<size_t> pos) const {
+inline std::string FunctionSchema::formatTypeMismatchMsg(
+    const Argument& expected,
+    const std::string& actual_type,
+    c10::optional<size_t> position,
+    c10::optional<std::string> value) const {
+  std::string position_str;
+  if (position) {
+    position_str = c10::str("Position: ", *position, "\n");
+  }
+  std::string value_str;
+  if (value) {
+    value_str = c10::str("Value: ", *value, "\n");
+  }
+  return c10::str(
+      name(),
+      "() ",
+      expected.formatTypeMismatchMsg(actual_type),
+      position_str,
+      value_str,
+      "Declaration: ",
+      *this);
+}
+
+inline void FunctionSchema::checkArg(
+    const IValue& value,
+    const Argument& argument,
+    optional<size_t> pos) const {
   if (!isSubvalueOf(value, argument.type())) {
     std::string position = pos ? ::c10::str(" in position ", *pos) : "";
-    AT_ERROR(
-        "Expected value of type ",
-        *argument.type(),
-        " for argument '",
-        argument.name(),
-        "'",
-        position,
-        ", but instead got value of type ",
-        attemptToRecoverType(value)->str(),
-        ". Declaration: ",
-        *this);
+    TORCH_CHECK(
+        false,
+        formatTypeMismatchMsg(
+            argument, attemptToRecoverType(value)->str(), pos));
```

Comment:
python_str() (a lot of our existing code has the problem, but the user should only ever see python_str())